client/testclient/cmd/register.go

```diff
@@ -0,0 +1,116 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"encoding/json"
+
+	"github.com/coniks-sys/coniks-go/client"
+	"github.com/coniks-sys/coniks-go/keyserver/testutil"
+	p "github.com/coniks-sys/coniks-go/protocol"
+	"github.com/spf13/cobra"
+)
+
+var registerCmd = &cobra.Command{
+	Use:   "register",
+	Short: "Register a name-to-key binding.",
+	Long: `Register a new name-to-key binding on the CONIKS-server.
+Note that, for the registration to work, the test-client needs to be running on the same machine as the key-server.
+A real client would first register an app here: https://apps.twitter.com (the UI should make this possible without actually leaving the client application). Then, the client would use the oauth1 API (find the tokens on the [Keys and Access Tokens] tab of the aforementioned page) and contact the (twitter-)bot for a registration request. The twitter bot is also accessible from outside.
+For more information consult: https://godoc.org/github.com/coniks-sys/coniks-go/bots
+
+Example call:
+  coniksclient register --name Alice@twitter --key fake_test_key`,
+	Run: func(cmd *cobra.Command, args []string) {
+		conf := loadConfigOrExit(cmd)
+		name := cmd.Flag("name").Value.String()
+		key := cmd.Flag("key").Value.String()
+		if len(name) == 0 || len(key) == 0 {
+			cmd.Usage()
+			return
+		}
+		msg, err := createRegistrationMsg(name, key)
+		if err != nil {
+			fmt.Println("Couldn't marshal registration request!")
+			os.Exit(-1)
+		}
+
+		addr, err := net.ResolveUnixAddr("unix", cmd.Flag("socket").Value.String())
+		if err != nil {
+			fmt.Println("Invalid socket address: " + err.Error())
+			os.Exit(1)
+		}
+		res, err := testutil.NewUnixClient(msg, addr)
+		if err != nil {
+			fmt.Println("Error while receiving response: " + err.Error())
+		}
+
+		response, errCode := client.UnmarshalResponse(p.RegistrationType,
+			res)
+		switch errCode {
+		case p.Success:
+			res, ok := response.(*p.DirectoryProof)
+			if !ok {
+				fmt.Println("Got unexpected response from server!")
+				os.Exit(-1)
+			}
+			// TODO why are the implementations for this verification
+			// method all empty? (see: protocol/message.go)
+			c := res.Verify(name, []byte(key), 0, nil,
+				conf.SigningPubKey)
+			// verify auth. path:
+			if c != p.Passed ||
+				res.AP == nil ||
+				!res.AP.Verify(res.STR.TreeHash) {
+				fmt.Println("Response message didn't pass verification (invalid auth. path).")
+			}
+			// verify signature on TB:
+			tbb := res.TB.Serialize(res.STR.Signature)
+			if !conf.SigningPubKey.Verify(tbb, res.TB.Signature) {
+				fmt.Println("Couldn't verify signature of temporary binding.")
+			}
+			fmt.Println("Succesfully registered name: " + name)
+			// TODO Where should the client save seen STRs and the TB?
+			// (or should it wait till the next epoch to see if the
+			// TB was actually inserted?)
```

Comment:
There maybe have more than one TB since the client could also receive the TB of her contacts, and we definitely have to check if the TB as been included as promised. Given that we will have a `ConiksClient` to store verified STRs and TBs (see https://github.com/coniks-sys/coniks-go/pull/74/commits/266d8cd3c55756d627e1e0581f8f3cc2ec3413a0#diff-75d470fe65b4ef49cd55654004d3edeaR29), I'm thinking that should we store both STRs and TBs in this struct?